modules/mono/build_scripts/build_assemblies.py

```diff
@@ -300,6 +306,9 @@ def main():
     parser.add_argument("--godot-platform", type=str, default="")
     parser.add_argument("--mono-prefix", type=str, default="")
     parser.add_argument("--push-nupkgs-local", type=str, default="")
+    parser.add_argument(
+        "--float", type=str, default="default", choices=["default", "32", "64"], help="Floating-point precision"
```

Comment:
The current code doesn't seem to be handling the "default" value. I think it should be something like this:

```py
if args.float == "default":
    args.float = "32"
```

But let's have @raulsntos confirm this.

Reply:
...plus `32` isnt doing anything in the code neither, so the only meaningful value is `64`, in which case it might be just a boolean switch like `--float64`?

Reply:
I put in the default value handling as it is more explicit. Without the handling it would just not use REAL_T_IS_DOUBLE but putting it in explicitly makes it clearer if the default is going to be changed engine-wide in future.

The float value has choices as that's consistent with how it is set when building the engine with scons.
It would be less friendly to vary it from the scons parameter (`scons float=64 p=linuxbsd etc`)

Reply:
The `default` doesn't seem to be used for anything in SConstruct either, so maybe it can be changed in both place to default to `32`.

Reply:
The `32` value doesn't need to set anything which results in `REAL_T_IS_DOUBLE` being unset. Since our default is the same as 32, `default` also doesn't need to do anything but I agree that making it more explicit by setting the argument to `32` is better, since now the rest of code only needs to check for the `32` or `64` values and can ignore `default` (although, just to be clear, since the way our code is written defaults to 32-bit types we don't need to do anything special in that case, only when the value is `64` do we need to set `REAL_T_IS_DOUBLE` to ensure the build uses 64-bit types instead).

I also agree that it's better to keep the argument as a string rather than boolean because that would allow us to potentially support other values in the future (e.g.: `128`) which is probably why it was implemented like this in SConstruct.

**TL;DR:** I agree with @neikeq's  and @akien-mga's suggestions.

Reply:
Actually, wouldn't it be better to remove the `default` option from both SConstruct & build_assemblies.py and just set the real default with the definition of the command-line parameters?

It looks a little weird to me to put 'default' as the default value and it just means the 'real' default value needs to be set elsewhere.

We could just do it like this in SConstruct:
```
opts.Add(EnumVariable("float", "Floating-point precision", "32", ("32", "64")))
```